File: Reverse a Linked List in groups of given size[21-July-2023]/Recursive/Solution.cpp
//  >> C++ RECURSIVE APPROACH <<

class Solution {
public:
    struct node* reverse(struct node* head, int k) {
        if (head == nullptr) return nullptr;
        
        struct node* prev = nullptr;
        struct node* next = nullptr;
        struct node* headptr = head;
        int cnt = k;
        
        while (headptr != nullptr && cnt != 0) {
            next = headptr->next;
            headptr->next = prev;
            prev = headptr;
            headptr = next;
            cnt--;
        }
        
        head->next = reverse(headptr, k);
        
        return prev;
    }
};

File: Reverse a Linked List in groups of given size[21-July-2023]/Recursive/Solution.java
//  >> JAVA RECURSIVE APPROACH

class Solution {
    public static Node reverse(Node head, int k) {
        if (head == null) return null;
        
        Node prev = null;
        Node next = null;
        Node headptr = head;
        int cnt = k;
        
        while (headptr != null && cnt != 0) {
            next = headptr.next;
            headptr.next = prev;
            prev = headptr;
            headptr = next;
            cnt--;
        }
        
        head.next = reverse(headptr, k);
        
        return prev;
    }
}